src/aspire/utils/blk_diag_matrix.py

```diff
@@ -16,6 +16,14 @@
 
 class BlkDiagMatrix:
 
+    # All instances of this class should have priority over ndarray ops
```

Comment:
Probably we should use more formal Docstring in multiple lines.

Reply:
Not sure what you mean here. There is a two line brief top level doc string, which is consistent with other classes in aspire code.  Every method, most importantly __init__, has a doc string.  Were you looking at the correct commit?

Reply:
I mean that these several lines of comments probably should be put as the general docstring for the `BlkDiagMatrix` class.

Reply:
I see.  That is not documentation for users, it is a note for other developers (including myself). I thought it was clear from the content, perhaps not.  This comment relates to the class code line directly below it, which is required for correct behavior, by a somewhat unusual consequence of combining ndarrays with operator overloading in python.

Regardless, it is not something that a user should know or care about, and doesn't belong in a user's documentation.  It is there to give context to developers working on that code that may not have experience with that problem in practice, or to remind myself at a later date why that line exists.  If I have to make any other changes, I will edit it for clarity.